src/TrackArtist.cpp

```diff
@@ -260,6 +260,68 @@ void TrackArtist::UpdatePrefs()
    SetColours(0);
 }
 
+void TrackArt::DrawClipAffordance(wxDC& dc, const wxRect& rect, bool highlight, bool selected, const wxString& title)
+{
+    // bottom rounded corners supposed to be hidden using clipping region or overdraw
+   if (selected)
+   {
+      wxRect strokeRect{
+         rect.x - ClipSelectionStrokeSize,
+         rect.y,
+         rect.width + ClipSelectionStrokeSize * 2,
+         rect.height + ClipFrameRadius };
+      dc.SetBrush(*wxTRANSPARENT_BRUSH);
+      AColor::UseThemeColour(&dc, clrClipAffordanceStroke, clrClipAffordanceStroke);
+      dc.DrawRoundedRectangle(strokeRect, ClipFrameRadius);
+   }
+   AColor::UseThemeColour(&dc, highlight ? clrClipAffordanceActiveBrush : clrClipAffordanceInactiveBrush, clrClipAffordanceOutlinePen);
+   dc.DrawRoundedRectangle(wxRect(rect.x, rect.y + ClipSelectionStrokeSize, rect.width, rect.height + ClipFrameRadius), ClipFrameRadius);
+   if (!title.empty())
+   {
+      dc.SetFont(wxFont(8, wxFONTFAMILY_SWISS, wxFONTSTYLE_NORMAL, wxFONTWEIGHT_NORMAL));
+      dc.SetTextForeground(*wxBLACK);
+      auto ascent = dc.GetFontMetrics().ascent;
+      dc.DrawText(title, rect.x + ClipFrameRadius, rect.y + (rect.height + ascent) / 2 - ascent);
```

Comment:
What happens if the text is so long that it extends past the clip's right boundary?

Another clip's affordance might be painted after over part of it.

The text is now always empty, but maybe you will need to truncate it with ...

Reply:
Also mentioned today:  What about right-to-left languages?  Maybe the code to draw the text needs to change in that case?

Reply:
Let solve this with PR related to clip names, until then i remove label from affordances